server/src/main/java/org/eclipse/lsp/cobol/service/delegates/references/ElementOccurrences.java

```diff
@@ -17,118 +17,107 @@
             .map(Context.class::cast)
             .map(ElementOccurrences::convertToElement)
             .map(ElementOccurrences::constructElementsExcludingImplicits);
-        return fromTree.orElseGet(() -> Stream.<Supplier<Optional<Element>>>of(
-                () -> findElementByPosition(result.getParagraphDefinitions(), result.getParagraphUsages(), position),
-                () -> findElementByPosition(result.getSectionDefinitions(), result.getSectionUsages(), position))
-
-            .map(Supplier::get)
-            .filter(Optional::isPresent)
-            .map(Optional::get)
-            .map(ElementOccurrences::constructElementsExcludingImplicits)
-            .findFirst().orElse(new Element(Collections.emptyList(), Collections.emptyList())));
-    }
-
-    private static Element constructElementsExcludingImplicits(Element e) {
-        List<Location> definitions = e.definitions.stream().filter(uriNotImplicit()).collect(Collectors.toList());
-        List<Location> usages = e.usages.stream().filter(uriNotImplicit()).collect(Collectors.toList());
-        return new Element(definitions, usages);
-    }
-
-    private static Predicate<Location> uriNotImplicit() {
-        return i -> !i.getUri().startsWith(PREF_IMPLICIT);
-    }
-
-    @Value
-    private static class Element {
-        List<Location> definitions;
-        List<Location> usages;
-    }
-
-    private static Optional<Element> findElementByPosition(
-            Map<String, List<Location>> definitions,
-            Map<String, List<Location>> usages,
-            TextDocumentPositionParams position) {
-        return Stream.concat(definitions.entrySet().stream(), usages.entrySet().stream())
-                .filter(entry -> entry.getValue().stream().anyMatch(location -> RangeUtils.isInside(position, location)))
-                .findFirst()
-                .map(Map.Entry::getKey)
-                .map(name -> new Element(
-                    definitions.getOrDefault(name, Collections.emptyList()),
-                    usages.getOrDefault(name, Collections.emptyList())));
-    }
-
-    private static Element convertToElement(Context contextNode) {
-        return new Element(convertToLocations(contextNode.getDefinitions()),
-            convertToLocations(contextNode.getUsages()));
-    }
-
-    private static List<Location> convertToLocations(List<Location> nodes) {
-        return nodes.stream()
-            .collect(Collectors.toList());
-    }
-
-    @NonNull
-    private static Predicate<Location> byUri(@NonNull TextDocumentPositionParams position) {
-        return location -> location.getUri().equals(position.getTextDocument().getUri());
-    }
 
-    @NonNull
-    private static Function<Location, DocumentHighlight> toDocumentHighlight() {
-        return location -> new DocumentHighlight(location.getRange(), DocumentHighlightKind.Text);
+    return fromTree.orElse(new Element(Collections.emptyList(), Collections.emptyList()));
```

Comment:
You create a new instance here, so it is better to you `orElseGet`

Reply:
done